kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-kogito-runtime/src/test/java/org/kie/workbench/common/stunner/kogito/client/selenium/BPMNDesignerKogitoSeleniumIT.java

```diff
@@ -0,0 +1,310 @@
+            } catch (IOException ioe) {
+                LOG.error("Unable to take screenshot", ioe);
+            }
+        }
+
+        @Override
+        protected void finished(Description description) {
+            if (driver != null) {
+                driver.quit();
+            }
+        }
+    };
+
+
+    @Test
+    public void testHandlingInvalidContent() {
+        setContent("<!!!invalid!!!>");
+
+        // Check that there is error modal dialog shown to user
+        final WebElement errorDialogModal = waitOperation().until(element(ERROR_MODAL_DIALOG));
+        assertThat(errorDialogModal)
+                .as("If invalid dmn is loaded, error dialog modal should be shown.")
+                .isNotNull();
+        // Check that there is a message explaining what happened
+        assertThat(errorDialogModal.findElement(By.xpath(ERROR_MODAL_BODY)).getText())
+                .as("If error dialog is shown, it show show proper message.")
+                .contains("Invalid BPMN file. Opening default text editor instead.");
+        errorDialogModal.click();
+
+        // Verify ACE editor (default text editor) is in place and shown to user
+        final WebElement aceEditor = waitOperation().until(element(ACE_EDITOR));
+        assertThat(aceEditor)
+                .as("If invalid dmn is loaded, ace editor needs to be shown")
+                .isNotNull();
+    }
+
+    @Test
+    public void testNewDiagram() throws Exception {
+        final String expected = loadResource("new-diagram.bpmn2");
+        setContent("");
+
+        final String actual = getContent();
+        assertThat(actual).isNotBlank();
+
+        // Skip, id, name and namespace in the comparison - they are dynamic
+        XmlAssert.assertThat(actual)
+                .and(expected)
+                .ignoreComments()
+                .ignoreWhitespace()
+                .withAttributeFilter(
+                        attr -> !(Objects.equals(attr.getName(), "id")
+                                || Objects.equals(attr.getName(), "name")
+                                || Objects.equals(attr.getName(), "namespace")))
+                .withNodeFilter(
+                        node -> !(Objects.equals(node.getNodeName(), "bpmn2:source")
+                                || Objects.equals(node.getNodeName(), "bpmn2:target")))
+                .areIdentical();
+    }
+
+    @Test
+    public void testBasicModel() throws Exception {
+        final String expected = loadResource("basic-process.bpmn2");
+        setContent(expected);
+
+
+        assertDiagramNodeIsPresentInProcessNavigator("Start");
+        assertDiagramNodeIsPresentInProcessNavigator("Add user to database");
+        assertDiagramNodeIsPresentInProcessNavigator("End");
+
+        final String actual = getContent();
+        assertThat(actual).isNotBlank();
+
+        XmlAssert.assertThat(actual)
+                .and(expected)
+                .ignoreComments()
+                .ignoreWhitespace()
```

Comment:
```suggestion
                .ignoreWhitespace()
                 .withAttributeFilter(
                        attr -> !(Objects.equals(attr.getName(), "id")
                        ))
                .withNodeFilter(
                        node -> !(Objects.equals(node.getNodeName(), "bpmn2:source")
                                || Objects.equals(node.getNodeName(), "bpmn2:target")))
```
@hasys tests can be fixed like this, if the `definition` element's `id` attribute is not valid for kogito